sources/Context.cpp

```diff
@@ -197,16 +273,59 @@ Context::execute() {
   simu_context->setWorkingDirectory(config_.outputDir().generic_string());
 
-  // This shall be the last log performed before building simulation,
-  // because simulation constructor will re-initialize traces for Dynawo
-  // Since DFL traces are persistent, they can be re-used after simulation is performed outside this function
-  LOG(info) << MESS(SimulateInfo, basename_) << LOG_ENDL;
+  switch (def_.simulationKind) {
+  case SimulationKind::STEADY_STATE_CALCULATION: {
+    // This shall be the last log performed before building simulation,
+    // because simulation constructor will re-initialize traces for Dynawo
+    // Since DFL traces are persistent, they can be re-used after simulation is performed outside this function
+    LOG(info) << MESS(SimulateInfo, basename_) << LOG_ENDL;
 
-  auto simu = boost::make_shared<DYN::Simulation>(jobEntry_, simu_context, networkManager_.dataInterface());
-  simu->init();
-  simu->simulate();
-  simu->terminate();
-  simu->clean();
+    // For a power flow calculation it is ok to directly run here a single simulation
+    auto simu = boost::make_shared<DYN::Simulation>(jobEntry_, simu_context, networkManager_.dataInterface());
+    simu->init();
+    simu->simulate();
+    simu->terminate();
+    simu->clean();
+    break;
+  }
+  case SimulationKind::SECURITY_ANALYSIS:
+    LOG(info) << MESS(SecurityAnalysisSimulationInfo, basename_, def_.contingenciesFilePath) << LOG_ENDL;
+    executeSecurityAnalysis();
+    break;
+  }
+}
+
+void
+Context::executeSecurityAnalysis() {
+  // For security analysis we run multiple simulations using dynawo-algorithms
+  // Create one scenario for the base case and one scenario for each contingency
+  auto scenarios = boost::make_shared<DYNAlgorithms::Scenarios>();
+  scenarios->setJobsFile(jobEntry_->getName() + ".jobs");
+  auto baseCase = boost::make_shared<DYNAlgorithms::Scenario>();
+  baseCase->setId("BaseCase");
+  scenarios->addScenario(baseCase);
+  if (validContingencies_) {
+    for (const auto& c : validContingencies_->get()) {
+      auto scenario = boost::make_shared<DYNAlgorithms::Scenario>();
+      scenario->setId(c.get().id);
+      scenario->setDydFile(basename_ + "-" + c.get().id + ".dyd");
+      scenarios->addScenario(scenario);
+      LOG(info) << MESS(ContingencySimulationDefined, c.get().id) << LOG_ENDL;
+    }
+  }
+  // Use dynawo-algorithms Systematic Analysis Launcher to simulate all the scenarios
+  auto multipleJobs = multipleJobs::MultipleJobsFactory::newInstance();
+  multipleJobs->setScenarios(scenarios);
+  auto saLauncher = boost::make_shared<DYNAlgorithms::SystematicAnalysisLauncher>();
+  saLauncher->setMultipleJobs(multipleJobs);
+  saLauncher->setOutputFile("sa.zip");
+  saLauncher->setDirectory(config_.outputDir().generic_string());
+  saLauncher->setNbThreads(config_.getNumberOfThreads());
+  saLauncher->init();
+  saLauncher->launch();
+#if _DEBUG_
+  saLauncher->writeResults();
```

Comment:
It was left to check this function from dynawo-algorithms. 

When the integration with Dynawo is performed without having to dump the final states as IIDM networks we will be interested in exchanging only aggregated results with other applications (PowSyBl).

I agree that this can be left out of this PR. I have removed the code and left a comment.

Reply:
In debug, we have control of the environment so we have access to the aggregate results without using an archive. And if we use powsybl, we use only release. So I don't understand what was your rational

Reply:
I was just testing the ability of dynawo-algorithms of creating the aggregate results.

In the future, even for release, it would be interesting to have only the final archive (zip) file exchanged with external applications.